var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -375,6 +375,11 @@ def install(self, spec, prefix):
                 filter_file('mpc++_r)', 'mpc++_r mpiFCC)', f, string=True)
                 filter_file('mpifc)', 'mpifc mpifrt)', f, string=True)
 
+    def setup_dependent_package(self, module, dependent_spec):
+        """Called before cmake packages's install() methods."""
+
+        module.cmake = Executable(self.spec.prefix.bin.cmake)
```

Comment:
If we're doing this, would it not also make sense to just remove the cmake assignment from `build_enviroment.py`?

Reply:
I hesitated on that lest it create fallout, but it's worth a shot.